compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1874,15 +1857,11 @@ mkDictCon :: Env -> TypeConApp -> [(LF.FieldName, LF.Expr)] -> LF.Expr
 mkDictCon env tcon fields
 -- Structs must have > 0 fields, therefore we simply make a typeclass a synonym for Unit
 -- if it has no fields.
-    | envLfVersion env `supports` featureTypeSynonyms && null fields = EUnit
-    | envLfVersion env `supports` featureTypeSynonyms = EStructCon fields
-    | otherwise = ERecCon tcon fields
+    | null fields = EUnit
+    | otherwise = EStructCon fields
 
 mkDictProj :: Env -> TypeConApp -> LF.FieldName -> LF.Expr -> LF.Expr
-mkDictProj env tcon =
-    if envLfVersion env `supports` featureTypeSynonyms
-      then EStructProj
-      else ERecProj tcon
+mkDictProj env _tcon = EStructProj
```

Comment:
I think we can go further with this one, 

```diff
- mkDictProj :: Env -> TypeConApp -> LF.FieldName -> LF.Expr -> LF.Expr
- mkDictProj env _tcon = EStructProj
+ mkDictProj :: LF.FieldName -> LF.Expr -> LF.Expr
+ mkDictProj = EStructProj
```

We probably do want to keep the synonym since it shows the intention is not merely projecting any structure, but specifically a class dict

Reply:
good catch 0db9facb4a872e847c1a9075969b4fa1efc89224